py/desitarget/io.py

```diff
@@ -2727,6 +2727,73 @@ def find_target_files(targdir, dr='X', flavor="targets", survey="main",
     return fn
 
 
+def read_mtl_svnveto_file(filename, survey="main"):
+    """Read an MTL veto file and perform basic checks on its format.
+
+    Parameters
+    ----------
+    filename : :class:`str`
+        The full path to the MTL veto file.
+    survey : :class:`str`, optional, defaults to "main"
+        To look up the right MTL data model. Examples might be``'main'``
+        or ``'svX``' (where X is 1, 2, 3 etc.) for the main survey and
+        different iterations of SV, respectively.
+
+    Returns
+    -------
+    :class:`~numpy.ndarray`
+        A structured numpy array of the MTL veto file.
+
+    Notes
+    -----
+        - The checks that are performed include:
+            - The file should include these columns (with correct type):
+                `RA`, `DEC`, `TARGETID`, `TIMESTAMP`
+            - No TARGETID should be duplicated.
+            - No TIMESTAMP lower down in the file should be earlier in
+              time that a TIMESTAMP higher up in the file.
```

Comment:
This function allows optional columns to exist as long as the required columns are there.  I think that is good, e.g. to support end-users including a "REASONS" column, or other parameters that were measured that led to the reason for the veto.  At the same time, this function returns the full original table including those extra columns, which might have datamodel implications downstream depending upon how this table is merged with others.

Consider allowing the existence of optional columns in the input file, but stripping the data to only the columns you expect in only the order you want them so that all downstream usage gets a consistent data model.  e.g. with
```python
return vetodata['RA', 'DEC', 'TARGETID', 'TIMESTAMP']
```